### PR TITLE
Ref task 4916 add audio to note

### PR DESCRIPTION
**Pull request by manhphan34**
Feature: Edit Screen
Link https://edu-redmine.framgia.vn/issues/4916

### CODE REVIEW

**Comment by manhphan34** [comment]
ready

File: app/src/main/java/framgia/com/mynote/screen/edit/MediaNoteUpdate.java [hunk 1]
```diff
@@ -1,48 +1,53 @@
...
-            }
+    private void play(String path) {
+        try {
+            mMedia.playAudio(path);
+            changeUIOfAudioImage();
+            return;
+        } catch (Exception e) {
+            mHandleClick.onPlayAudioFailed();
+        }
         mHandleClick.onPlayAudioFailed();
```
Comment (DoanVanToan) [inline app/src/main/java/framgia/com/mynote/screen/edit/MediaNoteUpdate.java]: duplicate listener

File: app/src/main/java/framgia/com/mynote/screen/edit/MediaNoteUpdate.java [hunk 2]
```diff
@@ -55,4 +60,10 @@ private void changeUIOfAudioImage() {
...
         mHandleClick.onStopAudio();
     }
-}
+
+    @Override
+    public void onCompletion(MediaPlayer mp) {
+        mHandleClick.onPlayAudio();
+        mMedia.reset();
+    }
+}
```
Comment (DoanVanToan) [inline app/src/main/java/framgia/com/mynote/screen/edit/MediaNoteUpdate.java]: add end of file

File: app/src/main/java/framgia/com/mynote/screen/edit/HandlerClick.java [hunk 3]
```diff
@@ -42,4 +50,8 @@
 
         void onChooseHourError(Exception e);
     }
-}
+
+    interface ItemAudioClickListener {
+        void onAudioItemSelected(Audio audio);
+    }
+}
```
Comment (DoanVanToan) [inline app/src/main/java/framgia/com/mynote/screen/edit/HandlerClick.java]: add end of file

File: app/src/main/java/framgia/com/mynote/screen/edit/MediaNoteUpdate.java [hunk 4]
```diff
@@ -1,48 +1,53 @@
...
-                changeUIOfAudioImage();
-                return;
-            } catch (Exception e) {
-                mHandleClick.onPlayAudioFailed();
-            }
+    private void play(String path) {
+        try {
+            mMedia.playAudio(path);
+            changeUIOfAudioImage();
+            return;
```
Comment (DoanVanToan) [inline app/src/main/java/framgia/com/mynote/screen/edit/MediaNoteUpdate.java]: remove this line

File: app/src/main/java/framgia/com/mynote/screen/edit/NoteUpdateActivity.java [hunk 5]
```diff
@@ -100,7 +111,53 @@ public void onPlayAudioFailed() {
...
+        mAudioCreatorDialog.dismiss();
+        showDialogSaveAudio(path);
+    }
+
+    private void showDialogSaveAudio(String path) {
+        AlertDialog.Builder builder = new AlertDialog.Builder(this);
+        builder.setMessage(R.string.message_record_audio_succes)
+                .setPositiveButton(R.string.button_save_title, (dialog, id)
+                        -> mUpdateViewModel.getAudio().setValue(path))
+                .setNegativeButton(R.string.button_cancle_title, (dialog, id) -> {
```
Comment (DoanVanToan) [inline app/src/main/java/framgia/com/mynote/screen/edit/NoteUpdateActivity.java]:
```suggestion
                .setNegativeButton(R.string.button_cancle_title, null)
```

File: app/src/main/java/framgia/com/mynote/screen/edit/NoteUpdateActivity.java [hunk 6]
```diff
@@ -317,4 +395,4 @@ public void onChooseHourError(Exception e) {
     private long getTime(long time) {
         return time + DateTimeUtil.TIME_COMPENSATION;
     }
-}
+}
```
Comment (DoanVanToan) [inline app/src/main/java/framgia/com/mynote/screen/edit/NoteUpdateActivity.java]: add end of file

**Comment by manhphan34** [comment]
ready

**Review by DoanVanToan — APPROVED** [review]

**Pull request merged**